### ISSUE TITLE
CA1602 Doesn't Understand Recursive Property Pattern for Non-Null Check

### ISSUE THREAD

**Issue by JasonBock** [dotnet/roslyn-analyzers#2878]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.5-beta1.final"

#### Diagnostic ID

CA1602

#### Repro steps

Before I added the analyzers, this code was fine:

```
public bool Equals(Cell other) =>
	other is { } && this.Location == other.Location &&
		this.Value == other.Value;
```

#### Expected behavior
Adding the analyzers shouldn't trigger CS1602 in this case.

#### Actual behavior
I get a warning saying that I need to validate `other`.

**Comment by mavasani** [comment]
New C# language feature that needs to be handled by the analyzer package.

**Comment by Youssef1313** [comment]
I think you're referring to `CA1062`.

@mavasani  This was already fixed and can be closed
